src/components/DropList/DropList.jsx

```diff
@@ -180,6 +187,8 @@ function DropListManager({
               item: itemToRemove,
               key: contentKey,
             })
+          } else if (isItemReset(selectedItem)) {
+            updatedSelection = selectedItems
```

Comment:
Good question! Let me see if I can answer somewhat clearly hehe

First it's useful to separate the DropList behaviour from the Table Chooser one.

DropList adds an item at the bottom of the list when `withResetSelectionItem` is added (like you I also don't love the naming but the reason I used it was to avoid adding a boolean + a string props, perhaps there is still a better name, specially because this item doesn't necessarily mean it adds any resetting functionality, or any other functionality for that matter).

An item gets appended that when clicked doesn't do _anything_ (but this doing nothing is what makes it special, because it behaves just like any other item except for being selected), and is of type `reset_droplist`. It's up to you what to do here, it's very likely you'll either need to "reset" (like in the column chooser case) or "clear selection" (can't really think of another use case and that is why "reset" didn't sound too bad in the end).

This leaves us with implementing the actual resetting on the column chooser, in `Table.ColumnChooser.jsx:25` the relevant bit is (modified for clarity here):

```jsx
<DropList
  onSelect={(selection, clickedItem) => {
    if (clickedItem.type === 'reset_droplist') {
        setDropListSelection(defaultColumns) 
         // In our case we are resetting, but you could do something like:
         // setDropListSelection(null)
      }
    }
  }
  selection={state.dropListSelection}
/>
```

> One thing to consider is that this means your DropList selection needs to be _controlled_ by using `selection` 

I hold on to a reference to the original shape of the columns here `Table.jsx:51` and use that to restore the columns. This reference is really only used for that and for setting the initial state, the columns we use for rendering are stored in state (`columns={state.columns}`).

Hope this helps!

Reply:
Thank you!